pkg/cloud/services/networking/network.go

```diff
@@ -91,6 +91,15 @@ func (s *Service) ReconcileNetwork(clusterName string, openStackCluster *infrav1
 	return nil
 }
 
+func (s *Service) DeleteNetwork(network *infrav1.Network) error {
+
```

Comment:
Can we change that code so it doesn't fail when the network already has been deleted?

Problem might be that we delete the network are somehow not able to patch the openstackcluster and the next reconcile fails because the network is already gone.